vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_embedding_info.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ml/pipeline/pipeline_embedding_info.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/time/time.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo() = default;
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo& EmbeddingPipelineInfo::operator=(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo::~EmbeddingPipelineInfo() = default;
+
+bool EmbeddingPipelineInfo::FromValue(const base::Value::Dict& root) {
+  if (absl::optional<int> value = root.FindInt("version")) {
+    version = value.value();
+  } else {
+    return false;
+  }
+
+  if (const auto* value = root.FindString("timestamp")) {
+    if (!base::Time::FromUTCString((*value).c_str(), &time)) {
```

Comment:
What is the provided format of the time in the `wtpwsrqtjxmfdwaymauprezkunxprysm` resource?

Reply:
`"timestamp": "2022-06-09 08:00:00.704847"`

Reply:
How about using ISO 8601 format? for the the time in the respource?

Reply:
@ptjames your thoughts on the above? Thanks

Reply:
i.e. `YYYY-MM-DDTHH:mm:ss.sssZ` same as you have however we only 3 decimal places for milliseconds